cmd/lakectl/cmd/local_commit.go

```diff
@@ -24,6 +32,47 @@ func findConflicts(changes local.Changes) (conflicts []string) {
 	return
 }
 
+func hasUncommittedOutsideLocalPrefix(ctx context.Context, client *apigen.ClientWithResponses, remote *uri.URI, idx *local.Index) bool {
+	currentURI, err := idx.GetCurrentURI()
+	if err != nil {
+		DieErr(err)
+	}
+
+	// Get first uncommitted change. If it's outside the local prefix, we're done
+	dirtyResp, err := client.DiffBranchWithResponse(ctx, remote.Repository, remote.Ref, &apigen.DiffBranchParams{
+		Amount: apiutil.Ptr(apigen.PaginationAmount(1)),
+	})
+	DieOnErrorOrUnexpectedStatusCode(dirtyResp, err, http.StatusOK)
+
+	if len(dirtyResp.JSON200.Results) == 0 {
+		return false
+	}
+	if slices.ContainsFunc(dirtyResp.JSON200.Results, func(diff apigen.Diff) bool {
+		return diff.PathType == "object" && !strings.HasPrefix(diff.Path, *currentURI.Path)
+	}) {
+		return true
+	}
+
+	// Get the first uncommitted change after the prefix. If it exists, we're also done
+	nextPrefix := fmt.Sprintf("%s%s", filepath.Clean(*currentURI.Path), asciiCharAfterSlash)
+	dirtyResp, err = client.DiffBranchWithResponse(ctx, remote.Repository, remote.Ref, &apigen.DiffBranchParams{
+		Amount: apiutil.Ptr(apigen.PaginationAmount(1)),
+		After:  apiutil.Ptr(apigen.PaginationAfter(nextPrefix)),
+	})
+	DieOnErrorOrUnexpectedStatusCode(dirtyResp, err, http.StatusOK)
+
```

Comment:
We can stop here instead of creating an unnecessary API call

Reply:
See the explanation in the previous comment's reply and let me know what you think.